File: src/main/java/com/bnisler/service/split/SplitService.java
package com.bnisler.service.split;

import com.bnisler.dao.SplitDao;
import com.bnisler.entity.Split;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.ArrayList;
import java.util.List;

@Service(value = "splitService")
@Transactional(readOnly = true)
public class SplitService {

    @Autowired
    private SplitDao splitDao;

    @Autowired
    private SplitMapper splitMapper;

    public List<SplitDetail> getAllSplits() {
        List<SplitDetail> splitDetails = new ArrayList<>();
        for (Split split : splitDao.findAllSplits()) {
            SplitDetail splitDetail = new SplitDetail();
            splitMapper.mapEntityToDetail(split, splitDetail);
            splitDetails.add(splitDetail);
        }
        return splitDetails;
    }

    public Split getSplit(Long splitId) {
        return splitDao.findSplitById(splitId);
    }

    public SplitDetail getSplitDetails(Long splitId) {
        Split split = splitDao.findSplitById(splitId);
        SplitDetail splitDetail = new SplitDetail();
        splitMapper.mapEntityToDetail(split, splitDetail);
        return splitDetail;
    }

    @Transactional
    public Long createSplit(SplitWriteRequest writeRequest) {
        Split split = new Split();
        splitMapper.mapToEntity(writeRequest, split);
        return splitDao.saveSplit(split);
    }

    @Transactional
    public void updateSplit(Long splitId, SplitWriteRequest writeRequest) {
        Split split = splitDao.findSplitById(splitId);
        splitMapper.mapToEntity(writeRequest, split);
        splitDao.updateSplit(split);
    }
}

File: src/main/frontend/app/components/split/split.service.js
(function () {
    'use strict';
    angular
        .module('assist-web')
        .factory('splitService', splitService);

    splitService.$inject = ['apiService'];

    function splitService(apiService) {

        return {
            getAllSplits: getAllSplits,
            getSplitDetails: getSplitDetails,
            updateSplit: updateSplit,
            createSplit: createSplit
        };

        function createSplit(writeRequest) {
            return apiService.post('/splits', writeRequest);
        }

        function updateSplit(splitId, writeRequest) {
            return apiService.put('/splits/' + splitId, writeRequest);
        }

        function getSplitDetails(splitId) {
            return apiService.get('/splits/' + splitId + '/details');
        }

        function getAllSplits() {
            return apiService.get('/splits');
        }
    }
})();